### PR TITLE
Require user-provided response function to be instances 

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description

Continues the implementation of only allowing rfuncs to be provided as Instances (ps.One()) instead of classes, as requested and started per #360. If users provide a rfunc class instead of an instance, an Error will be thrown asking the user to provide an instance. 
 
EDIT: still work I progress, but should be done 17.01.2023 9:00, ready for review.

# Checklist before PR can be merged:
- [x] part of issue #360
- [x] is documented
- [x] PEP8 compliant code
- [x] tests added / passed
- [ ] Example Notebook (for new features)

### CODE REVIEW

**Comment by raoulcollenteur** [comment]
I think this PR should be good to go. Change also to be documenten as per #421.

**Review by dbrakenhoff — APPROVED** [review]

**Review by martinvonk — APPROVED** [review]
I searched for any leftover rfunc classes but didn't find any so seems good to me.

File: pastas/stressmodels.py [hunk 1]
```diff
@@ -694,7 +693,7 @@ def __init__(
             isinstance(rfunc, HantushWellModel) or issubclass(rfunc, HantushWellModel)
```
Comment (martinvonk) [inline pastas/stressmodels.py:693]: Same comment here as for the issubclass check on line 1424 @davidbrakenhoff mentioned. I believe only the isinstance check is enough. `issubclass(class, classinfo)` returns True if a class is a subclass of classinfo. The subclass here would be RfuncBase for classinfo HantushWellModel (and Exponential). And we don't need to check for RfuncBase right?

Comment (raoulcollenteur) [reply pastas/stressmodels.py]: Agreed, removed.

File: pastas/stressmodels.py [hunk 2]
```diff
@@ -1422,7 +1421,7 @@ def __init__(
         if rfunc is None:
             rfunc = Exponential()
         if not (isinstance(rfunc, Exponential) or issubclass(rfunc, Exponential)):
```
[the remark this answers is not in the document]
Comment (raoulcollenteur) [reply pastas/stressmodels.py]: Agreed, removed.

**Comment by raoulcollenteur** [comment]
Thanks for the quick review, made the suggested changes and I will merge as soon as GH actions finishes (positively). @rubencalje no need for the review anymore :) 

Cheers,
Raoul

**Pull request merged**